Restructured to make installable via pip install

carlbalmer opened the pull request:
moved package into its own directory
moved examples into its own directory
added setup.py and pyproject.toml
modified import statements to match new structure

I tried to make as little changes as possible but decided to move the root python package into its own sub-directory (as it is usually done in python packaging). Leaving it in the root would have required non standard and hacky setup.py config.

NeilBurch commented:
Hi,

Installing helps avoid a bunch of the path shenanigans loading the pieces of the environment library - I like that!


There are a few issues.

`python rl_env_example.py` does not work unless one changes to the examples directory.  A small thing, but exactly following the README commands should hopefully work. Updating the README to suggest using `python examples/rl_env_example.py` seems like a 100% reasonable fix.

`python game_example.py` does not work as it needs numpy. It also has the same path issue as rl_env_example.py.

The C++ example in examples/game_example.cc is no longer built.


Suggest changing setup.py description to be 'Learning environment' rather than 'Reinforcement learning environment'. We might love RL, but we would like to include other things too.


`hanabi_learning_environment/agents/rainbow/run_experiment.py` could also be updated to `from hanabi_learning_environment import rl_env` to make use of your changes to a locally installed hanabi environment package.

carlbalmer commented:
> `python game_example.py` does not work as it needs numpy

Added instructions to install numpy to the README. I did not add it to setup.py to be installed automatically as it is not required by the learning environment.

> The C++ example in examples/game_example.cc is no longer built.

I added build instructions in the examples directory. I did not add it to the main CMakeLists.txt as it is not required by the learning environment and installing it to site-packages makes no sense.

NeilBurch commented:
Thank you.
